ops/charm.py

```diff
@@ -449,13 +449,11 @@ def __init__(self, raw: dict = {}, actions_raw: dict = {}):
         self.series = raw.get('series', [])
         self.subordinate = raw.get('subordinate', False)
         self.min_juju_version = raw.get('min-juju-version')
-        self.requires = {name: RelationMeta('requires', name, rel)
```

Comment:
I don't feel these read very well. It is a "new provides relation" but that really just rolls off the tongue `</sarcasm>`.

I could live with this, as still better than what we have, but ultimately I prefer the Enum route.

Reply:
Agreed, this the one I less like of the three.